dds/DCPS/ReactorTask.cpp

```diff
@@ -272,6 +323,31 @@ String ThreadStatus::get_key(const char* safety_profile_tid, const String& name)
   return key;
 }
 
+bool ThreadStatusManager::sync_with_parent(ThreadStatusManager& parent,
+  ThreadStatusManager::Map& running, ThreadStatusManager::Map& finished)
+{
+  ACE_READ_GUARD_RETURN(ACE_Thread_Mutex, g1, parent.lock_, false);
+  ACE_WRITE_GUARD_RETURN(ACE_Thread_Mutex, g2, lock_, false);
+
+  // Check for finished threads
+  for (Map::iterator i = map_.begin(); i != map_.end(); /* Do in loop to support erase */) {
+    if (parent.map_.count(i->first) == 0) {
```

Comment:
I don't know how big these maps are, but since we're locking two mutexes here this might be a good candidate for walking two iterators over both maps (as sorted "lists") rather than using std::map::count in order to move from O(n*log(m)) to O(n+m).

Reply:
They are the threads that OpenDDS is using, so they're normally tiny, like 4. I can sorta see how I could use two iterators to do it, especially because I'm trying turn one map into the other while recording the difference. I think it'd be more complicated though to try to keep track of where the two iterators in relation to each other.